packages/fx-core/src/component/utils/common.ts

```diff
@@ -55,3 +55,21 @@ export function asFactory<T>(keyValidators: KeyValidators<T>) {
     throw PrerequisiteError.somethingIllegal("Deploy", "data", "plugins.bot.InvalidData");
   };
 }
+
+// Expand environment variables in content. The format of referencing environment variable is: ${{ENV_NAME}}
+export function expandEnvironmentVariable(content: string): string {
+  const placeholderRegex = /\${{[a-zA-Z_][a-zA-Z0-9_]*}}/g;
+  const placeholders = content.match(placeholderRegex);
+
+  if (placeholders) {
+    for (const placeholder of placeholders) {
+      const envName = placeholder.slice(3, -2); // removes `${{` and `}}`
```

Comment:
maybe we should add .trim() to avoid the user using the environment variables like `${{  xxx }}`?

Reply:
Perhaps we can use `toUpperCase()` here to normalize env var names.

Reply:
@Siglud Thanks for the suggestion. Added trim to allow white spaces around env name. 
@1yefuwang1 though env name should be upper case, users can indeed set env name with lower case characters. So my preference is respecting the name provided by user.